### DISCUSSION TITLE
Using GloDroid on Orange Pi PC

### DISCUSSION THREAD

**Discussion by plambe74** [GloDroid/glodroid_manifest#125] [General]
Dear Project Owner,
First of all, thank you very much for your amazing work with Orange PC boards.
I was not really sure where to put my feedback, so I decided to start a general discussion.
Let me know if that belong more to the issues section.

I have an Orange PC, which I want to use ideally with Android TV Android flavour.
I am willing to contribute with my experience in bring up GloDroid on my device.

So far I have used your main instructions and build all images.
I have created SD card using the command bellow:
sudo dd if=deploy-sd.img of=/dev/sdb  bs=4k count=300000
I have then run from the image directory the command bellow:
 ./flash-sd.sh 
It seemed to run fine apart from the following mesage:
 ./fastboot flash super super.img
Invalid sparse file format at header magic
It continued nevertheless and my device was rebooted.
At the end I ended in what I think is half-loaded Android, putting on a message:
"There is an internal problem with your device, contact your manifacturer for details". 
I clicked on the OK button and ended up with a running os. 

I have good experience with building Linux for embedded devices with the help of Yocto, OpenWRT and Buildroot.
It is my first experience building on android.

I am interesting in making that board work with the USB Wifi Dongle D-Link DWA 131.
Is that dongle already supported?

If you can advise on a good How To link for system changes on Adroid, I will be happy.

In fact I have older Android for the same board and would be interested to build a ko module for that dongle that I can use with that ROM as well.

Thank you very much for taking your time to read my feedback.
I will be happy to get an answer and be useful to your project.

All the best, stay safe and keep the good work with the OrangePi boards.
Best regards,
Plambe

**Comment by plambe74** [comment]
Dear rcglobal,
Does Android tv flavour work well on Orange Pi PC board? 

Do you know, if the board is recognised by Android? 

I see that it is not possible to use Google Play with whatever board, or am I wrong? 

Same goes with chrome cast. 
Have you experimented with this feature? 
Is the Android TV flavour operational and stable with latest build on Orange Pi PC? 

I see that build is using Android 11 now. 
Could you please answer my questions? 

I see that rsglobal is active on other threads. 

So I see that you use latest android, but that becomes pretty limitted w/o Google Play and w/o the possibility to add a Google acount. 

Thank you for taking your time to read and answer.

**Reply by rsglobal** [reply]
Hi,

AOSP has limited functionality comparing to Android.
Using Google Services is impossible (without additional tricks). Unfortunately I have no experience with Chromecast.

Orange PI PC has 1 GB of RAM, which is very low to run functional AOSP-11.
You can probably use it to run single app, which doesn't require heavy launchers and rich UI.

I can't tell you much about Android-TV variant, last time it was tested 1.5 years ago. Android-TV variant is only a custom launcher with black background and simplified settings menu, nothing more.

**Reply by rsglobal** [reply]
> I am interesting in making that board work with the USB Wifi Dongle D-Link DWA 131.

You can add corresponding kernel config:

Example: https://github.com/GloDroid/glodroid_device/blob/1fe99ca9dd51119b07ce1e162b040b673f7c22fe/opi3/kernel.config#L2

**Reply by plambe74** [reply]
Rsglobal, Thank you for your answer.
I see that my RAM could be an issue. 

Is it possible then to build from your code base older Android version. 
I know from the vendor website that the Opi PC was tested with Android 7.
Will the build work, if I provide e. g.
branch: android-7.1.2_r36
to the repo init command? 

As for the "internal error problem..."  it may be that the /system/build.prop and the /vendor/build.prop values don’t match.
Perhaps this is due to the device not being certified for android 11 or is some error in the build and install scripts. 

By the way chromecast is very nice feature. 
I am looking around for an open source variant of it. I found something that worked on RPi. It may be portable on the OPi as well. 

At the end if I can use Kodi decent browser and some open phone screen mirroring, I will not need Google play services.
![IMG_20210119_173944](https://user-images.githubusercontent.com/36604579/105064686-59446b00-5a7d-11eb-9a76-0cf60f6eabcb.jpg)


Thanks again for your feedback.

**Reply by plambe74** [reply]
Dear rsglobal, Could you please answer if you build older versions of AOSP? 

I see that almost all OrangePi boards come with limitted RAM. 

Are you interested in maintaining device specific configurations that work stable on the OPi platforms? 

My goal is more in this direction. 

And I like in fact your focus on AOSP and F-droid as market place.

**Reply by rsglobal** [reply]
Hi,

Using older version of Android is not possible out of the box. I do not recommend to go this direction.
Google had project called Android-Go, it aims to cover low-ram systems by relying on go instead of Java, but the project fate is unknown.

> As for the "internal error problem..." 

It is some bug in the system configuration or because of low ram some of services has been killed. It is not a certification issue.

> By the way chromecast is very nice feature.

I agree, I am using it all the time on my Xiaomi smart-tv. ) . Is it opensource?

> Are you interested in maintaining device specific configurations that work stable on the OPi platforms?
Currently our main goad is to collect generic set of low-level components (HALs) and build configuration to run latest Android on any board which can run Linux.

Currently we do not have real customer/product, so stabilization isn't important for us as for now.

But anyway, we can support you wherever our interests are aligned.
